data/remnant missions.txt

```diff
@@ -649,66 +644,135 @@ mission "Remnant: Catalytic Ramscoop"
 	on complete
 		outfit "Catalytic Ramscoop" -2
 		payment 1500000
-		log "People" "Tali" `As an engineering prefect, Tali is the foremost authority among the Remnant on salvaged technology and a senior leader in their reverse engineering program.`
 		conversation
 			`You visit the local shipyard and eventually find the engineer busily repairing a damaged Starling. You inform her that you have brought a pair of catalytic ramscoops for her to study. She quickly tightens a couple more connectors and jumps down from her perch, gesturing at an assistant to transfer you the credits you were promised. "Thank you! Our maps of human space are old, and if they are still accurate, you traveled a very long way to bring these to us." Her chanting voice echoes off the Starling above you with an interesting resonance.`
 			`	"If we can unlock the secrets of these ramscoops, it will give our ships greater range as they search for a safer refuge." As she picks up the remote control for the small loading truck with the two ramscoops she trills at you over her shoulder. "Oh, my name is Tali. I'll let you know what I find."`
 
 
 
-mission "Remnant: Plasma Cannons"
-	name "Retrieve Plasma Cannons for the Remnant"
-	description "An engineer on <planet> has offered to pay you <payment> credits in exchange for obtaining two plasma cannons."
+mission "Remnant: Plasma Cannon Check"
+	name "Check to see if player has a Plasma Cannon"
 	minor
 	source
 		government "Remnant"
 	to offer
 		has "Remnant: Technology Available: offered"
-		not "Remnant: Catalytic Ramscoop: active"
-		not "Remnant: Catalytic Ramscoop: done"
-		not "Remnant: Heavy Laser: active"
-		not "Remnant: Heavy Laser: done"
+	invisible
+	on offer
+		require "Plasma Cannon"
+	to complete
+		has "Remnant: Plasma Weapons: offered"
+
+
+
+mission "Remnant: Plasma Turret Check"
+	name "Check to see if player has a Plasma Turret"
+	minor
+	source
+		government "Remnant"
+	to offer
+		has "Remnant: Technology Available: offered"
+	invisible
+	on offer
+		require "Plasma Turret"
+	to complete
+		has "Remnant: Plasma Weapons: offered"
+
+
+
+mission "Remnant: Plasma Weapons"
+	name "Retrieve Plasma Cannons for the Remnant"
+	description "An engineer on <planet> has offered to pay you <payment> in exchange for obtaining two plasma cannons."
+	source
+		government "Remnant"
+	to offer
+		has "Remnant: Technology Available: offered"
+		or
+			has "Remnant: Plasma Cannon Check: offered"
+			has "Remnant: Plasma Turret Check: offered"
 		random < 50
 	on offer
 		require "Plasma Cannon"
```

Comment:
This mission still requires a plasma cannon, meaning it won't offer even if you have a plasma turret.